### PR TITLE
completed assessment

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -15,46 +15,164 @@
...
+        // Expected output: [1, 1, 2, 3, 5, 8]
 
+        // Example input: 10
+        // Expected output: [1, 1, 2, 3, 5, 8, 13, 21, 34, 55]
 
+      expect(fibSeq(6)).toEqual([1, 1, 2, 3, 5, 8])
+      expect(fibSeq(10)).toEqual([1, 1, 2, 3, 5, 8, 13, 21, 34, 55])
+    })
+  })
+//jest test fails because "ReferenceError: fibSeq is not defined"
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:32]: perfect

File: code-challenges.test.js [hunk 2]
```diff
@@ -15,46 +15,164 @@
...
+        //adds two previous values together before this index and pushes to fibArray
+        fibArray.push(fibArray[i-1]+fibArray[i-2]); 
+    }
+    return fibArray;
+}
+//originally failed test trying to use a map method but realized it doesn't update array values until the function has completed. Switched to if/else and for loops and test passed
+
+
+//I wanted to try recursion! ==========================  (it ended up being messier than most) (ノ• ◡•)ノ 彡┻━┻ 
+
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:54]: Amazing ascii

File: code-challenges.test.js [hunk 3]
```diff
@@ -15,46 +15,164 @@
...
+//fibSeq(3) is added to the call stack, and returns array = fibSeq(2)
+//fibSeq(2) is added to the call stack, since count = 2, we execute else if statement and returns array = fibSeq(1)
+//fibSeq(1) is added to the call stack, since count = 2, we execute else if statement and returns array = fibSeq(0)
+//this is where we hit the base case and where we find out fibSeq(0) returns []
+//we now work last in first out on the call stack
+//at fibSeq(1) we know array = [] so we push 1 into the array, it returns [1]
+//at fibSeq(2) array = [1] so we push another 1 into the array, it returns [1, 1]
+//at fibSeq(3) array = [1, 1] so we push the addition of (array[0] + array[1]) OR (1+1), returns [1, 1, 2]
+//at fibSeq(4) array = [1, 1, 2] we push (array[1] + array[2]) OR (1+2), returns [1, 1, 2, 3]
+//at fibSeq(5) array = [1, 1, 2, 3] we push (array[2] + array[3]) OR (2+3), returns [1, 1, 2, 3, 5]
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:95]: This is a great work up of recursion!

File: code-challenges.test.js [hunk 4]
```diff
@@ -15,46 +15,164 @@
...
 
+//create fn onlyOdd that takes array param
+const onlyOdd = (array) =>{
+    //filters array to return only odd values AND only with typeof "number"
+    let oddsArray = array.filter(value =>{ 
+        return value % 2 !== 0 && typeof value === "number"
+    });
+
+    //returns the sorted oddsArray from smallest to largest where if a-b < 0 sort a before b
+    return oddsArray.sort((a, b) => a-b);
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:128]: Excellent work!

File: code-challenges.test.js [hunk 5]
```diff
@@ -15,46 +15,164 @@
...
+}
+//failed test because it seems filter does type coercion regardless of the strictly not equals and because sort fn will sort based on strings if no comparison is identified. Passed test after fixing it. Can't really think of a way this might need refactoring.
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:130]: The only thing here would be appending the dot notation into one line. 
```javascript 
return array.filter(value => return value % 2 !== 0 && typeof value === "number").sort((a, b) => a-b);
```

File: code-challenges.test.js [hunk 6]
```diff
@@ -15,46 +15,164 @@
...
+    } 
+    
+    //otherwise continue through code block to for loop to iterate through inputArray and push to new array with summed values
+    for(let i = 0; i < inputArray.length; i++){
+        sum += inputArray[i]; //tracks overall sum
+        sumArray.push(sum); //pushes the compounding sum to each position in the array
+    }
+    return sumArray;
+}
+//test passed! couldn't really find anything that may need refactoring.
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:178]: I agree! great work!

File: interview-questions.md [hunk 7]
```diff
@@ -9,59 +9,58 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 4. What is JSX? What is one notable difference between HTML and JSX?
 
-  Your answer:
+  Your answer: JSX is JavaScript XML which is the code we use in the render() method within react. It is sort of like HTML.
 
-  Researched answer:
+  Researched answer: As above, JSX refers to JavaScript XML. It is a syntax extension of JavaScript which allows us to write HTML directly into React but still gives us the same powers of JavaScript. It allows us to combine logic and markup in one file.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:37]: eXcellent!

File: interview-questions.md [hunk 8]
```diff
@@ -9,59 +9,58 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 5. What is yarn? What file(s) are modified in a React application when you run yarn?
 
-  Your answer:
+  Your answer: yarn stands for "yet another resource negotiator". the command yarn will update your node modules folder to what is needed for the project.
 
-  Researched answer:
+  Researched answer: yarn is a new package manager. It updates our node modules folder so that we may avoid rewritting basic component, libraries, or frameworks of our own. It manages code dependencies. It was created to replace the already existing package manager npm. It differs in that it can handle multiple packages at once but npm can only do one by one. yarn is considered more reliable, faster, and more secure than npm.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:45]: Yes! So began the package manager wars

File: react-debugging.html [hunk 9]
```diff
@@ -79,6 +80,7 @@ <h1>Draw a Card</h1>
                   )
                 })}
               </div>
+            </>
```
Comment (WalkerTexas-coder) [inline react-debugging.html:83]: Great work here!

**Review by WalkerTexas-coder — APPROVED** [review]
What can I say Rebecca? Great work! You did a solid job of explaining all of the tools you used and then some. Your code is readable as is even without pseudo code (the long long term goal) and your tests are exact. Your interview answers are all solid and I really appreciate your deep dives. This assessment is perfectly on point! Keep up the good work! this PR is approved

File: code-challenges.test.js (hunk 5 above)
Comment (ghost) [reply code-challenges.test.js:130]: Thanks! I appreciate this feedback.

**Pull request merged**